### PR TITLE
feat/task-3: Create Home page with the list of transactions

### PR DESCRIPTION
**Pull request by JediDevYury**
#3 #4

### CODE REVIEW

**Review by angelinabylitskaya — COMMENTED** [review]
1. It is better not to attach login to icon names from berndor libraries (to avoid complicated changes on design changes)
2. category colors and names are better be stored in one object with category name
3. Consider reusable style theme (config or design system atoms, molecules)

File: api/firestone.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { db } from '@/services/firebase.service';
+import { User } from 'firebase/auth';
+import { addDoc, collection, doc, getDoc, setDoc, writeBatch } from 'firebase/firestore';
```
Comment (angelinabylitskaya) [inline api/firestone.ts]: please re-name file to firestore

File: app/(auth)/sign-in.tsx [hunk 2]
```diff
@@ -33,7 +34,7 @@ export default function SignIn() {
       />
       <Input isPassword placeholder="Password" value={password} onChangeText={setPassword} />
       <Button text="Sign In" onPress={handleSignIn} />
-      <Text className="mt-4">
+      <Text style={Typography.regular}>
```
Comment (angelinabylitskaya) [inline app/(auth)/sign-in.tsx]: `<Text regular>` can be used instead

File: services/firebase.service.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-const app = initializeApp(firebaseConfig);
 
+const app = initializeApp(firebaseConfig);
 const auth = initializeAuth(app, {
   persistence: getReactNativePersistence(ReactNativeAsyncStorage),
 });
+const db = getFirestore(app);
 
-export { app, auth };
+export { app, auth, db };
```
Comment (angelinabylitskaya) [inline services/firebase.service.ts:13]: it is not necessery to export these objects, instead getAuth(), getApp(), getFirestore() can be used

File: shared/Title.tsx [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
 }: PropsWithChildren<TextProps & TitleProps>) => {
   return (
-    <Text className={className || 'font-firaSemibold text-2xl mb-4'} {...props}>
+    <Text style={[styles.title, style]} {...props}>
       {text || children}
     </Text>
   );
 };
+
+const styles = StyleSheet.create({
```
Comment (angelinabylitskaya) [inline shared/Title.tsx]: title styles can be defined in react-native-ui-lib typography config and used as `<Text title>..`

File: app/(main)/index.tsx [hunk 5]
```diff
@@ -1,17 +1,57 @@
...
+        <Text>There are no transactions to display.</Text>
+      </View>
+    );
+  }
 
   return (
-    <View className="flex-1 justify-center items-center">
-      <Text onPress={onSignOut}>Sign Out</Text>
+    <View style={styles.container}>
+      <TransactionsList transactions={transactions} categories={categories} />
```
Comment (angelinabylitskaya) [inline app/(main)/index.tsx]: TransactionsList is not a re0usable component, get transaction logic can be moved inside. Home component can have any information

**Review by angelinabylitskaya — COMMENTED** [review]
Please consider moving shared components inside components folder, shared configs separate from components

File: services/firebase.service.ts (hunk 3 above)
Comment (JediDevYury) [reply services/firebase.service.ts:13]: I use those functions with params in different places, and for that it helps me not to create a new instance every time. That it's why for me this approach more preferable.

**Review by angelinabylitskaya — APPROVED** [review]

File: app/(main)/components/TransactionsList.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  const {
+    transactions = [],
+    isFetchingNextPage,
+    hasNextPage,
+    error: transactionsError,
+    loadMoreTransactions,
+  } = useTransactions();
+  const errorMessage = categoriesError?.message || transactionsError?.message;
+
+  const getCategory = (id: string) => categories?.find((category) => category.id === id);
```
Comment (angelinabylitskaya) [inline app/(main)/components/TransactionsList.tsx]: it is better to prepare data before to optimise performance

File: scripts/reset-project.js [hunk 7]
```diff
@@ -3,7 +3,7 @@
 
 /**
  * This script is used to reset the project to a blank state.
- * It moves the /app directory to /app-example and creates a new /app directory with an index.tsx and _layout.tsx file.
+ * It moves the /app directory to /app-example and creates a new /app directory with an sign-in.tsx and _layout.tsx file.
```
Comment (angelinabylitskaya) [inline scripts/reset-project.js:6]: this script can be removed already from the project

**Review by angelinabylitskaya — APPROVED** [review]

**Pull request merged**